clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -688,6 +722,7 @@ public void abortTransaction() throws ProducerFencedException {
      * @param callback A user-supplied callback to execute when the record has been acknowledged by the server (null
      *        indicates no callback)
      *
+     * @throws IllegalStateException if a transactional.id has been configured and no transaction has been started
```

Comment:
Hmm, we mention in other methods two possible reasons why `UnsupportedVersionException` can be thrown:

1. Broker is too old.
2. Offsets topic is configured to use an older message format.

What if the topic you are sending to is configured with an older message format?